### DISCUSSION TITLE
enter_room & leave_room in multiple nodes

### DISCUSSION THREAD

**Discussion by gus3inov** [miguelgrinberg/python-socketio#988] [Q&A] [answered]
Continuing the discussion that I recently created https://github.com/miguelgrinberg/python-socketio/discussions/973.
 I have server with multiple nodes and there is using enter_room & leave_room functions.
 Consider we have 3 nodes: node A (nA), node B (nB), node C (nC).

nA executed `enter_room` function for `sid_user_1` to room `bot_agent_1`:
```python
sio_server.enter_room(
    sid=sid_user_1,
    room=bot_agent_1,
    namespace=GAME_NAMESPACE,
)
```

nB executed `enter_room` function for `sid_user_1` to room `bot_agent_2`:
```python
sio_server.enter_room(
    sid=sid_user_1,
    room=bot_agent_2,
    namespace=GAME_NAMESPACE,
)
```

Then, nC executes `emit` function for `bot_agent_1` and `bot_agent_2`

```python
sio_sever.emit(
    key,
    response,
    namespace=GAME_NAMESPACE,
    room=bot_agent_1,
)

sio_sever.emit(
    key,
    response,
    namespace=GAME_NAMESPACE,
    room=bot_agent_2,
)
```

Will the client `sid_user_1` receive sockets with rooms bot_agent_1 and bot_agent_2 that were sent from nodes nA and nB?
I assume that the sid -> user information is stored in-memory and this information is not transferred between pods in any way

P.S I will be grateful for any help in these matters!

**Comment by miguelgrinberg** [comment, marked as answer]
Unfortunately this is currently not supported. You can only call `enter_room` in an event handler for the affected user. If you enter a user to a room from a node that does not know the user then the call will be ignored.

**Reply by gus3inov** [reply]
Thanks!

**Comment by gus3inov** [comment]
@miguelgrinberg There is the way to make the other instance give information about the `sid` via `save_session`?
If I execute `save_session(sid_user_1)` on node nA and nB will it work?

**Reply by miguelgrinberg** [reply]
No. Sessions are stored in the memory of the process, they are not shared across processes. An interesting extension would be to allow for more session store options, but at this time only memory is implemented.

**Reply by gus3inov** [reply]
Ok, got it. I'll try to add PR soon with support for storing this in redis
